app/views/clientsmodal.html

```diff
@@ -1,126 +1,83 @@
-            <i tooltip-placement="top" uib-tooltip="A password has been set for this client" class="glyphicon glyphicon-lock"></i>
-          </label>
+<div class="modal-body">
+  <div class="form-group has-error-key">
+    <span>*</span> Indicates a required field<br />
+    <span>*</span><span>*</span> Indicates that at least one of the fields is required
+  </div>
 
-          <label><span>**</span> Basic Auth Password</label>
-          <input class="form-control" placeholder="Set/change password..." type="password" ng-model="temp.password">
-          <div class="has-error-msg"><i class="glyphicon glyphicon-warning-sign"></i>Either '<strong>Certificate</strong>' or '<strong>Password</strong>' is required!</div>
-        </div>
-        <div class="form-group" style="float: right;" ng-class="{ 'has-error' : ngError.passwordConfirm }">
-          <label>Confirm Password</label>
-          <input class="form-control" placeholder="Confirm password..." type="password" ng-model="temp.passwordConfirm">
-          <div class="has-error-msg"><i class="glyphicon glyphicon-warning-sign"></i>{{validationPasswordConfirmMsg}}</div>
-        </div>
-      </div>
+  <fieldset>
+    <uib-tabset justified="true">
+      <!-- ************* TAB FOR CLIENT BASIC INFO ************** -->
+      <uib-tab
+        class="{{ngError.clientBasicInfoTab === true ? 'has-error-tab' : ''}}"
+        active="selectedTab.basicInfo"
+      >
+        <uib-tab-heading
+          ><i
+            class="glyphicon glyphicon-warning-sign"
+            style="display: none;"
+          ></i
+          >Basic Info</uib-tab-heading
+        >
+        <h3>Basic Info</h3>
+        <div
+          ng-include="'views/partials/clients-tab-basic-info.html'"
+        ></div>
+      </uib-tab>
 
-      <div class="form-group" ng-class="{ 'has-error' : ngError.clientDomain }">
-        <label>Domain</label>
-        <input class="form-control" type="text" ng-model="client.clientDomain" >
-        <div class="has-error-msg"><i class="glyphicon glyphicon-warning-sign"></i>{{validationRequiredMsg}}</div>
-      </div>
-      <div class="form-group" ng-class="{ 'has-error' : ngError.certFingerprint }">
-        <label><span>**</span> Client certificate</label>
-        <select class="form-control" ng-options="cert.fingerprint as 'cn=' + cert.commonName + ', o=' + cert.organization for cert in certs" ng-model="client.certFingerprint">
-          <option value="" label="No client certificate"></option>
-        </select>
-        <div class="has-error-msg"><i class="glyphicon glyphicon-warning-sign"></i>Either '<strong>Certificate</strong>' or '<strong>Password</strong>' is required!</div>
-      </div>
+      <!-- ************* TAB FOR Client Authentication ************** -->
+      <uib-tab
+        class="{{ngError.clientAuthenticationTab === true ? 'has-error-tab' : ''}}"
+        active="selectedTab.authentication"
+      >
+        <uib-tab-heading
+          ><i
```

Comment:
I think this closing bracket should be on the same line as the tag above. There arent any attributes supplied to that element and doesnt need to be split on a new line